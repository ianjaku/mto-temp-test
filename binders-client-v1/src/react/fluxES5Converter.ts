/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
