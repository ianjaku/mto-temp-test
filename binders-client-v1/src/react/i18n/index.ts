import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
