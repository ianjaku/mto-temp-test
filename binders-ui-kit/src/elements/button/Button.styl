@require "../../vars.styl";
@require "../../style/global.styl";

makeButtonDisabled()
    opacity: 0.2
    cursor: default

.button
    font-family: defaultFontName, sans-serif
    color: whiteColor
    background-color: baseColor
    padding: 10px 34px
    border-radius: 5px
    display: inline-flex
    align-items: center
    justify-content: center
    text-align: center
    transition: all 0.2s ease-in-out
    font-size: 13px
    font-weight: 500
    margin: 5px 0 5px 0
    position: relative
    text-decoration: none
    line-height: 1em
    border: 1px solid baseColor
    @media rwd.tabletAndAbove
        padding: 10px 40px
    &:not(.button--disabled)&:not(.button--unhoverable)
        cursor: pointer
        &:hover
            background-color: accentColor
            border-color: accentColor
    &:focus
        outline: none
    &--disabled
        makeButtonDisabled()
    &--secondary
        background-color: transparent
        border-color: baseColor
        color: baseColor
        &.branded
            border-color: var(--mainColor)
            color: var(--mainColor)
        &:not(.button--disabled)&:not(.button--unhoverable)
            &:hover
                border-color: baseColor
                background-color: baseColor
                color: white
                &.branded
                    border-color: var(--mainColor-hover)
                    background-color: var(--mainColor-hover)
        &^[0]--disabled
            makeButtonDisabled()
    &--CTA
        background-color: accentColor
        border-color: accentColor
        &.branded
            background-color: var(--mainColor)
            border-color: var(--mainColor)
        &:not(.button--disabled)&:not(.button--unhoverable)
            &:hover
                box-shadow: 0px 2px 3px 0px rgba(0,0,0,0.35)
                &.branded
                    background-color: var(--mainColor)
                    border-color: var(--mainColor)
        &^[0]--disabled
            makeButtonDisabled()
    &--borderless
        border: none
    &-round
        width: 24px;
        height: 24px;
        border-radius: 15px;
        background-color: accentColor;
        color: whiteColor;
        border: none;
        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
        cursor: pointer;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        img
            box-shadow: none
        &:focus
            outline: none;

        .material-icons
            font-size: 20px !important;
            color: whiteColor !important;
    &-loader
        right: 10px
        top: 8px
        color: whiteColor !important
        &--secondary
            color: baseColor !important

    &-link
        position: absolute
        display: block
        top: 0
        bottom: 0
        right: 0
        left: 0

