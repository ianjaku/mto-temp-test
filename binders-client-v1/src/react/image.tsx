/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
