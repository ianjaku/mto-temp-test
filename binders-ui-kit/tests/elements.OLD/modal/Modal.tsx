import * as React from "react";
import Button from "../../../src/elements/button";
import Modal from "../../../src/elements/modal";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const TITLE = "Vegetables & fruit";
const CONTENT = "Buy here! You're favourite fruit and vegetables";

function empty() {
    return;
}

const createEmptyModal = () => wrapWithTheme(<Modal />);
const createHiddenModal = () => wrapWithTheme(<Modal hidden={true} />);
const createModalWithTitle = title => wrapWithTheme(<Modal title={title} />);
const createModalWithContentAndTitle = (title, content) => (
    wrapWithTheme(<Modal title={title}>
        <p>{content}</p>
    </Modal>)
);
const createModalWithButtons = () => {
    const buttons = [wrapWithTheme(<Button text="apple" onClick={empty} />), wrapWithTheme(<Button text="cherry" onClick={empty}/>)];
    return wrapWithTheme(<Modal buttons={buttons}>Save changes?</Modal>);
};
const createModalWithoutPadding = () => (
    wrapWithTheme(<Modal withoutPadding={true}>
        <table className="showcase-modal-table">
            <tr>
                <th>Color</th>
                <th>Fruits</th>
            </tr>
            <tr>
                <td>Red</td>
                <td>Apples, cherries, ...</td>
            </tr>
            <tr>
                <td>Green</td>
                <td>Green apples</td>
            </tr>
            <tr>
                <td>Orange</td>
                <td>oranges, mandarines, ...</td>
            </tr>
        </table>
    </Modal>)
);

// most preferred
describe("Modal", () => {
    test("Empty modal (snapshot)", () => {
        const modal = create(createEmptyModal());
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Empty hidden modal (snapshot)", () => {
        const modal = create(createHiddenModal());
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Modal with title (snapshot)", () => {
        const modal = create(createModalWithTitle(TITLE));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Modal with title and content with padding (snapshot)", () => {
        const modal = create(createModalWithContentAndTitle(TITLE, CONTENT));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Modal with title and table without padding (snapshot)", () => {
        const modal = create(createModalWithoutPadding());
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Modal title (mount: enzyme+jsdom)", () => {
        const elementText = mount(createModalWithTitle(TITLE))
            .find(".modal-header > h3")
            .at(0)
            .text();
        expect(elementText).toEqual(TITLE);
    });

    test("Modal title (mount: enzyme+jsdom)", () => {
        const elementText = mount(createModalWithContentAndTitle(TITLE, CONTENT))
            .find(".modal-body > p")
            .at(0)
            .text();
        expect(elementText).toEqual(CONTENT);
    });

    test("Modal buttons (mount: enzyme+jsdom)", () => {
        const buttons = mount(createModalWithButtons()).find(".modal-footer > ul > li").length;
        expect(buttons).toEqual(2);
    });
});
