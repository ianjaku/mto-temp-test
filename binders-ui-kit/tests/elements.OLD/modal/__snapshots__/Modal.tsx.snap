// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Empty hidden modal (snapshot) 1`] = `null`;

exports[`Modal Empty modal (snapshot) 1`] = `
<div
  className="modal-wrapper"
>
  <div
    className="modal"
  >
    <header
      className="modal-header"
    >
      <h3
        className="modal-header-title"
      />
      <button
        className="close"
        onClick={[Function]}
      >
        <span
          className="material-icons "
          color="inherit"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "inherit",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          close
        </span>
      </button>
    </header>
    <div
      className="modal-body"
    />
  </div>
</div>
`;

exports[`Modal Modal with title (snapshot) 1`] = `
<div
  className="modal-wrapper"
>
  <div
    className="modal"
  >
    <header
      className="modal-header"
    >
      <h3
        className="modal-header-title"
      >
        Vegetables & fruit
      </h3>
      <button
        className="close"
        onClick={[Function]}
      >
        <span
          className="material-icons "
          color="inherit"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "inherit",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          close
        </span>
      </button>
    </header>
    <div
      className="modal-body"
    />
  </div>
</div>
`;

exports[`Modal Modal with title and content with padding (snapshot) 1`] = `
<div
  className="modal-wrapper"
>
  <div
    className="modal"
  >
    <header
      className="modal-header"
    >
      <h3
        className="modal-header-title"
      >
        Vegetables & fruit
      </h3>
      <button
        className="close"
        onClick={[Function]}
      >
        <span
          className="material-icons "
          color="inherit"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "inherit",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          close
        </span>
      </button>
    </header>
    <div
      className="modal-body"
    >
      <p>
        Buy here! You're favourite fruit and vegetables
      </p>
    </div>
  </div>
</div>
`;

exports[`Modal Modal with title and table without padding (snapshot) 1`] = `
<div
  className="modal-wrapper"
>
  <div
    className="modal"
  >
    <header
      className="modal-header"
    >
      <h3
        className="modal-header-title"
      />
      <button
        className="close"
        onClick={[Function]}
      >
        <span
          className="material-icons "
          color="inherit"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "inherit",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          close
        </span>
      </button>
    </header>
    <div
      className="modal-body modal-body--withoutPadding"
    >
      <table
        className="showcase-modal-table"
      >
        <tr>
          <th>
            Color
          </th>
          <th>
            Fruits
          </th>
        </tr>
        <tr>
          <td>
            Red
          </td>
          <td>
            Apples, cherries, ...
          </td>
        </tr>
        <tr>
          <td>
            Green
          </td>
          <td>
            Green apples
          </td>
        </tr>
        <tr>
          <td>
            Orange
          </td>
          <td>
            oranges, mandarines, ...
          </td>
        </tr>
      </table>
    </div>
  </div>
</div>
`;
